11-payment-encoding.md

```diff
@@ -0,0 +1,339 @@
+SHOULD fail if `amount` contains a non-digit, or is followed by
+anything except a `multiplier` in the table above.
+
+A reader SHOULD indicate if amount is unspecified, otherwise it MUST
+multiply `amount` by the `multiplier` value (if any) to derive the
+amount required for payment.
+
+## Rationale
+
+The `amount` is encoded into the human readable part, as it's fairly
+readable and a useful indicator of how much is being requested.
+
+Donation addresses often don't have an associated amount, so `amount`
+is optional in that case: usually a minimum payment is required for
+whatever is being offered in return.
+
+# Data Part
+
+The data part consists of multiple sections:
+
+1. `version`: 0 (5 bits)
+1. `timestamp`: seconds-since-1970 (35 bits, big-endian)
+1. `payment_hash`: (256 bits + 4 trailing zero bits)
+1. Zero or more tagged parts.
+1. `signature`: bitcoin-style signature of above. (520 bits)
+
+## Requirements
+
+A writer MUST set `version` to 0, MUST set `timestamp` to the time to
+the number of seconds since Midnight 1 January 1970, UTC in
+big-endian, and MUST set `payment_hash` to the SHA-2 256-bit hash of
+the `payment_preimage` which will be given in return for payment,
+followed by four zero bits.  A writer MUST set `signature` to a valid
+512-bit secp256k1 signature of the double SHA2 256-bit hash of the
+Human Readable Part concatenated with a bytes for each 5 bits of the
+Data Part, with a trailing byte containing the recovery ID (0, 1, 2 or
+3).
+
+## Rationale
+
+The 5-bit `version` fits neatly into base32 encoding (0 == `q`), and
+allows for future incompatible upgrades.  `signature` covers the
+pre-base32 encoded data for simplicity, especially since the data part
+may not end on an 8-bit boundary.  It also allows public key recovery,
+so the identity of the payee node is implied.
+
+## Tagged Fields
+
+Each Tagged Field is of format:
+
+1. `type` (5 bits)
+1. `length` (10 bits, big-endian)
+1. `data` (length x 5 bits)
+
+Currently defined Tagged Fields are:
+
+* `d` (13): short description of purpose of payment,  e.g. '1 cup of coffee'
+* `h` (23): 256-bit description of purpose of payment (SHA256).  This is used to commit to an associated description which is too long to fit, such as may be contained in a web page.
+* `x` (6): `expiry` time in seconds (big-endian). Default is 3600 (1 hour) if not specified.
+* `f` (9): fallback on-chain address.  For bitcoin, this is 5 bits of
+      witness version followed by a witness program, with witness
+      version `3` (17) meaning P2PKH, `j` (18) meaning [P2SH](https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki); both are followed by a 20-byte hash value.
+* `r` (3): extra routing information.  This should be appended to the route
+      to allow routing to non-public nodes; there may be more than one of these.
+   * `pubkey` (264 bits)
+   * `channel_id` (64 bits)
+   * `fee` (64 bits, big-endian)
+   * `cltv_expiry_delta` (16 bits, big-endian)
+
```

Comment:
This is not consistent with the description for the "r" tagged field description in the [simplified description](https://github.com/rustyrussell/lightning-payencode) which says 32-bit fee and 32-bit cltv. Which is correct?